src/main/java/org/elasticsearch/common/geo/GeometryCollection.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.common.geo;
+
+import com.spatial4j.core.context.SpatialContext;
+import com.spatial4j.core.shape.Rectangle;
+import com.spatial4j.core.shape.Shape;
+import com.spatial4j.core.shape.ShapeCollection;
+import com.spatial4j.core.shape.SpatialRelation;
+
+import java.util.Collection;
+import java.util.List;
+
+import static com.spatial4j.core.shape.SpatialRelation.CONTAINS;
+import static com.spatial4j.core.shape.SpatialRelation.INTERSECTS;
+
+/**
+ *  Overrides bounding box logic in ShapeCollection base class to comply with
+ *  OGC OpenGIS Abstract Specification: An Object Model for Interoperable Geoprocessing.
+ *
+ *  This class also overrides the 'relate' method to leverage the updated bbox logic.
+ *  NOTE: This algorithm is O(N) and can possibly be improved O(log n) using an internal R*-Tree
+ *  data structure for a collection of bounding boxes
+ */
+public class GeometryCollection<S extends Shape> extends ShapeCollection<S> {
```

Comment:
I think this should be XSpaceCollection if these changes will be going back into spatial4j?